llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1948,6 +2045,28 @@ void AsmPrinter::emitFunctionBody() {
         break;
       }
 
+      // FIXME: Some indirect calls can get lowered to jump instructions,
+      // resulting in emitting labels for them. The extra information can
+      // be neglected while disassembling but still takes space in the binary.
+      if (TM.Options.EmitCallGraphSection && MI.isCall()) {
+        // Only indirect calls have type identifiers set.
+        const auto &CallSiteInfo = CallSitesInfoMap.find(&MI);
```

Comment:
```suggestion
        const auto &CallSiteInfo = CallSitesInfoMap.find(&MI);
      if(CallSiteInfoe == CaleSitesInfoMap.end())
        return;
```
Maybe not syntactically correct here, but I feel like there should be some early exit here that reduces nesting or maybe this is a helper function that handles some of this complexity?

Reply:
I've extracted this snippet into a new method to clean up the code. PTAL.